File: S04/T04/prob05.cpp
/*
 * Created on Tue Jul 26 11:42:20 2022

 * @author: Gerardo A. Rivera Tello
 * @handle: DangoMelon
 */

#include <iostream>

int main(){
    int n, curr = 1, prev = 0;

    std::cout << "Ingrese la cantidad de terminos a imprimir de la sucesion de Fibonacci: ";
    std::cin >> n;

    for (int i = 0; i < n; i++)
    {
        std::cout << prev << " ";
        int temp = curr;
        curr += prev;
        prev = temp;
    }
    std::cout << std::endl;
}

File: S04/T04/prob05.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jul 20 15:19:33 2022

@author: Gerardo A. Rivera Tello
@handle: DangoMelon
"""

num = eval(
    input("Ingrese la cantidad de terminos a imprimir de la sucesion de Fibonacci: ")
)

prev = 0
curr = 1
for i in range(num):
    print(prev, end=" ")
    prev, curr = curr, prev + curr
print()